server.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"encoding/json"
+	errs "errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type ServerInfo struct {
+	BuildDate time.Time `json:"build_date"`
+	BuildSha  string    `json:"build_sha"`
+	Version   string    `json:"version"`
+}
+
+func (info ServerInfo) String() string {
+	if info.Version == "" {
+		return "unknown apm-server version"
+	}
+	return fmt.Sprintf("apm-server version %s built on %d %s [%s]",
+		info.Version, info.BuildDate.Day(), info.BuildDate.Month().String(), info.BuildSha[:7])
+}
+
+func QueryServerInfo(secret, url string) (ServerInfo, error) {
```

Comment:
I'd prefer this is added in a separate pull request since it's not related to the custom transport